File: CodingSample/DotNet/Fundamentals/Inheritance/InterfaceTypeTest/BankLib/SavingsAccount.cs
namespace Banking;

sealed class SavingsAccount : Account, IProfitable
{
    //an identifier declared with 'const' keyword is replaced
    //with the value assigned to it whereever it occurs in code 
    const decimal MinBal = 5000;

    internal SavingsAccount()
    {
        Balance = MinBal;
    }

    public override void Deposit(decimal amount)
    {
        Balance += amount;
    }

    public override void Withdraw(decimal amount)
    {
        if(Balance - amount < MinBal)
            throw new InsufficientFundsException();
        Balance -= amount;
    }

    public void AddInterest(int months)
    {
        decimal rate = Balance < 5 * MinBal ? 3 : 4;
        decimal interest = Balance * rate * months / 1200;
        Balance += interest;
    }
}

File: CodingSample/Java/Core/Inheritance/InterfaceTest/bank/banking/SavingsAccount.java
package banking;

final class SavingsAccount extends Account implements Profitable {
    
    //a field declared with 'final' keyword
    //cannot be re-assigned
    static final double MIN_BAL = 5000;

    SavingsAccount() {
        balance = MIN_BAL;
    }

    public void deposit(double amount) {
        balance += amount;
    }

    public void withdraw(double amount) throws InsufficientFundsException {
        if(balance - amount < MIN_BAL)
            throw new InsufficientFundsException();
        balance -= amount;
    }

    public double interest(int months) {
        double rate = balance < 5 * MIN_BAL ? 3 : 4;
        return balance * rate * months / 1200;
    }
}
